src/foam/nanos/auth/email/EmailVerificationWebAgent.js

```diff
@@ -94,20 +83,14 @@ foam.CLASS({
           translatedMsg = ts.getTranslation(local, getClassInfo().getId()+ ".EMAIL_VERIFIED_ERROR", this.EMAIL_VERIFIED_ERROR);
           message = translatedMsg + "<br>" + msg;
         } finally {
-          if ( getConfig() == null ) {
-            setConfig(EnvironmentConfigurationBuilder
-                .configuration()
-                .resources()
-                .resourceLoaders()
-                .add(new TypedResourceLoader("dao", new DAOResourceLoader(x, (String) user.getGroup())))
-                .and().and()
-                .build());
-          }
-    
+          EmailTemplateEngine templateEngine = (EmailTemplateEngine) x.get("templateEngine");
+          templateEngine.setX(x);
+          HashMap args = new HashMap();
+          args.put("msg", message);
+          templateEngine.setValues(args);
```

Comment:
What if some other thread is running at the same time and sets the values to something else?
Couldn't we pass this as another argument to renderrTemplate()?

Reply:
yes, I see